### PR TITLE
Fix crash in RSVP sending

### PR DESCRIPTION
**Pull request by bitfriend**
When sending RSVP, app crashes.
Will fix it.
image

### CODE REVIEW

**Comment by bitfriend** [comment]
@gnunicorn explained me about it.

>I recommend looking closely at all types handed out in the API calls around the crashing calls, see if any of them might be references or expecting referenced types ...
>I had the a similar problem in tasks using `description()` not returning a clone and so rust would discard the object after and any further call of that function or object would return garbled memory ...

**Comment by bitfriend** [comment]
`RsvpDraft::status()` returns `&mut Self` on rust side.
But it should return `void` on flutter side.
Otherwise app crashes when calling of `RsvpDraft::status()` on flutter side.

File: native/acter/api.rsh [hunk 1]
```diff
@@ -387,7 +387,7 @@ object RsvpManager {
 
 object RsvpDraft {
     /// set status of this RSVP
-    fn status(status: string) -> RsvpDraft;
+    fn status(status: string);
```
Comment (gnunicorn) [inline native/acter/api.rsh:390]: doesn't that still have an `unreachable` internally?

**Review by gnunicorn — APPROVED** [review]
I'm fine with merging this as is, but there is still an unreachable we need to remove, too.

File: native/acter/api.rsh (hunk 1 above)
Comment (gnunicorn) [reply native/acter/api.rsh:394]: as said, let's merge as-is

File: native/acter/src/api/rsvp.rs [hunk 2]
```diff
@@ -75,19 +78,32 @@ pub struct RsvpDraft {
...
 impl RsvpDraft {
     pub fn status(&mut self, status: String) -> &mut Self {
-        let Ok(s) = RsvpStatus::from_str(&status) else {
-            unreachable!("Wrong status about RSVP")
-        };
-        self.inner.status(s);
+        if let Ok(s) = RsvpStatus::from_str(&status) {
+            self.inner.status(s);
+        } else {
+            error!("Wrong status about RSVP");
```
Comment (gnunicorn) [inline native/acter/src/api/rsvp.rs:84]: yeah, that's fine.

**Pull request merged**